### PR TITLE
feat(title): make it possible to specify the title

### LINKED ISSUE #11: More useful title
Just the path to the file isn't very useful if it doesn't include which repo it is pulled from.

Can we have it use the full URL? I'm guessing it can't be an actual link.

At the very least maybe we can toggle it on/off or allow another argument?

### PR DESCRIPTION
**Pull request by rryter**
Hello there

I'm not familiar with react, so if I did this totally wrong, let me know 👍 

## Usage

> \```js reference title="Instantiation"
> https://github.com/ERC725Alliance/erc725.js/blob/feature/simplify-api/examples/decodeData/index.js#L1-L45
> \```

Screenshot 2021-08-03 at 14 09 43

Closes #11

### CODE REVIEW

File: src/theme/ReferenceCodeBlock/index.tsx [hunk 1]
```diff
@@ -111,7 +118,7 @@ function ReferenceCode(props: ReferenceCodeBlockProps) {
 
     const customProps = {
         ...props,
-        metastring: ` title="${codeSnippetDetails.title}"`,
+        metastring: getTitle(props, codeSnippetDetails),
```
Comment (christian-bromann) [inline src/theme/ReferenceCodeBlock/index.tsx:121]: I don't think we need an extra function for this.
```suggestion
        metastring: props.title ? ` title="${props.title.replace('"', '')}"` : ` title="${codeSnippetDetails.title}"`,
```

**Comment by rryter** [comment]
I just realised this will not work with spaces in the title, e.g. "Title One" 

This will result in "Title"

I'm looking into it.

**Comment by rryter** [comment]
> I just realised this will not work with spaces in the title, e.g. "Title One"
> This will result in "Title"
> I'm looking into it.

I resolved it, let me know what you think.

Thanks a lot. 👍

**Review by christian-bromann — APPROVED** [review]
Thank you 👍

**Pull request merged**